### ISSUE TITLE
False positive for Lint/UselessAssignment and Lint/UnusedMethodArgument

### ISSUE THREAD

**Issue by obfuscoder** [bbatsov/rubocop#5580]
With the recent updater of parser to version 2.5.0.0, Rubocop reports a false positive for the last argument for a method which has a variable argument list and a block. It's difficult to put into words. Please see the example below.

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.5.0.0, running on ruby 2.4.1 x86_64-linux)
```

Example:

```
module Prawn
  class Document
    def bounding_box(pt, *args, &block)
      # ...
    end
  end
end

def test
  left = 10
  top = 20
  cell_width = 30
  cell_height = 40
  bounding_box [left, top], width: cell_width, height: cell_height do
    stroke_bounds
    barcode label
  end
end
```

Result:

```
test.rb:..:..: W: Lint/UselessAssignment: Useless assignment to variable - cell_height.
    cell_height = 40
    ^^^^^^^^^^^
```

The false possitive is not reported when putting the arguments within parenthese when calling the method:

```
  # ...
  bounding_box([left, top], width: cell_width, height: cell_height) do
  # ...
```

If this last parameter is an input argument to the test method, it will be reported as Lint/UnusedMethodArgument.

This false positive was not reported, when using parser 2.4.0.2.

**Comment by mikegee** [comment]
Although it doesn't look like it, I think this might be a dupe of https://github.com/bbatsov/rubocop/issues/5573. See the explanation of the bug in [Parser](https://github.com/whitequark/parser/issues/447#issuecomment-366712218).

**Comment by Darhazer** [comment]
Can't reproduce this in Rubocop 0.54, using Parser 2.5.1.0
The referenced parser issue is closed as fixed. Perhaps we can close this issue as well
